internal/eru/resources/gpu.go

```diff
@@ -0,0 +1,245 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	_ "embed"
+
+	"github.com/kr/pretty"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/jaypipes/ghw"
+	"github.com/projecteru2/core/log"
+	"github.com/projecteru2/yavirt/configs"
+	"github.com/projecteru2/yavirt/internal/types"
+	"github.com/projecteru2/yavirt/internal/vmcache"
+	gputypes "github.com/yuyang0/resource-gpu/gpu/types"
+)
+
+var execCommand = exec.Command
+
+var (
+	//go:embed gpu_name.json
+	gpuNameJSON string
+	gpuNameMap  = map[string]string{}
+)
+
+type SingleTypeGPUs struct {
+	gpuMap  map[string]types.GPUInfo
+	addrSet mapset.Set[string]
+}
+
+type GPUManager struct {
+	mu           sync.Mutex
+	gpuTypeMap   map[string]*SingleTypeGPUs
+	coreMgr      *CoreResourcesManager
+	lostGPUCache *cache.Cache
+}
+
+func initGPUNameMap(gpuProdMapCfg map[string]string) error {
+	if len(gpuNameMap) > 0 {
+		return nil
+	}
+	if err := json.Unmarshal([]byte(gpuNameJSON), &gpuNameMap); err != nil {
+		return err
+	}
+	// merge name map from config to gpuNameMap
```

Comment:
`lo.Map` would help?

Reply:
I think for-loop is very intuitive here, using functional style code here don't have any advantages.